app/instance-initializers/google-tag-manager.js

```diff
@@ -0,0 +1,15 @@
+import config from '../config/environment';
+
+export function initialize(/* appInstance */) {
+  if (typeof window !== 'undefined' && typeof document !== 'undefined' && config.environment !== 'test') {
```

Comment:
you can shorten this to 
```suggestion
  if (typeof FastBoot === 'undefined' && config.environment !== 'test') {
```